internal/cri/server/container_create.go

```diff
@@ -362,6 +372,119 @@ func (c *criService) CreateContainer(ctx context.Context, r *runtime.CreateConta
 	return &runtime.CreateContainerResponse{ContainerId: id}, nil
 }
 
+func (c *criService) mutateMounts(
+	ctx context.Context,
+	platform imagespec.Platform,
+	extraMounts []*runtime.Mount,
+) error {
+	for _, m := range extraMounts {
+		err := c.mutateOCIBindMount(ctx, platform, m)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *criService) mutateOCIBindMount(
+	ctx context.Context,
+	platform imagespec.Platform,
+	extraMount *runtime.Mount,
+) (retErr error) {
+	imageSpec := extraMount.GetImage()
+	if imageSpec == nil {
+		return nil
+	}
+	if extraMount.GetHostPath() != "" {
+		return fmt.Errorf("hostpath must be empty while mount image: %+v", extraMount)
+	}
+	if !extraMount.GetReadonly() {
+		return fmt.Errorf("readonly must be true while mount image: %+v", extraMount)
+	}
+
+	ref := imageSpec.GetImage()
+	if ref == "" {
+		return fmt.Errorf("image not specified in: %+v", imageSpec)
+	}
+
+	image, err := c.LocalResolve(ref)
+	if err != nil {
+		return fmt.Errorf("failed to resolve image %q: %w", ref, err)
+	}
+
+	containerdImage, err := c.toContainerdImage(ctx, image)
+	if err != nil {
+		return fmt.Errorf("failed to get image from containerd %q: %w", image.ID, err)
+	}
+
+	// This is a digest of manifest
+	id := containerdImage.Target().Digest.Hex()
+	target := c.getImageVolumeDir(id)
```

Comment:
maybe call this the imageVolumeHostPath :) or something less than final target .. cached volume .. ivd ?